crates/typst-cli/src/compile.rs

```diff
@@ -170,9 +170,16 @@ fn export_pdf(document: &Document, command: &CompileCommand) -> StrResult<()> {
     Ok(())
 }
 
+/// Get time passed as environment variable
+fn passed_time() -> Option<NaiveDateTime> {
+    let passed_string = option_env!("SOURCE_DATE_EPOCH")?;
+    let epoch = passed_string.parse::<i64>().ok()?;
+    chrono::DateTime::from_timestamp(epoch, 0).map(|date| date.naive_utc())
```

Comment:
Invalid values in `SOURCE_DATE_EPOCH` should probably result in a warning or error rather than silently using wall time instead. If the env variable is set, the intent is clearly for it to be used as well.